### ISSUE TITLE
Error while producing catTuple from RelVal sample

### ISSUE THREAD

**Issue by jhgoh** [vallot/CATTools#146]
"GenMETExtractor" module tries to read slimmedMETs::PAT but the slimmedMET was produced with "RECO" process name.

```
Begin processing the 1st record. Run 1, Event 1, LumiSection 1 at 10-Aug-2015 15:53:40.883 CEST
----- Begin Fatal Exception 10-Aug-2015 15:53:50 CEST-----------------------
An exception of category 'LogicError' occurred while
   [0] Processing run: 1 lumi: 1 event: 1
   [1] Running path 'outpath'
   [2] Calling event method for module PoolOutputModule/'out'
   [3] Calling produce method for unscheduled module CATMETProducer/'catMETs'
   [4] Calling produce method for unscheduled module PATMETSlimmer/'slimmedMETs'
   [5] Calling produce method for unscheduled module CorrectedPATMETProducer/'patPFMetT1'
   [6] Calling produce method for unscheduled module PATMETProducer/'patPFMet'
   [7] Calling produce method for unscheduled module GenMETExtractor/'genMetExtractor'
Exception Message:
Hit circular dependency while trying to run an unscheduled module.
The last module on the stack shown above requested data from the
module with label: 'slimmedMETs'.
This is illegal because this module is already running (it is in the
stack shown above, it might or might not be asking for data from itself).
More information related to resolving circular dependences can be found here:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideUnscheduledExecution#Circular_Dependence_Errors.
----- End Fatal Exception -------------------------------------------------

```

Adding

```
process.genMetExtractor.metSource = cms.InputTag("slimmedMETs", "", "RECO")
```

solves the problem but this is just a temporary solution.
Any suggestions?

**Comment by geonmo** [comment]
I think that this suggestion is best for us. 
Could you make a PR for this?

**Comment by jhgoh** [comment]
can we assume slimmedMETS::RECO for all samples? The production worked with slimmedMETs::PAT right?

**Comment by geonmo** [comment]
Well, I think that this feature can be handled with MiniAOD option at cmsRun option.
Is it wrong?

**Comment by jshlee** [comment]
the problem is that prompt uses reco and rereco uses PAT...

looking at the problem.. patPFMet shouldnt be calling slimmedMETs
i'll have a look into it.

**Comment by jhgoh** [comment]
This issue have been standing quite long time.
Thinking about use case, we usually use RelVal samples only for the testing, thus we can workaround by checking substring in the input file(s) and switch to use slimmedMETs::RECO.

Actually this hack to the PAT2CAT_cfg.py was already included (by accident?) in #191.
Case closed.
